flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -184,7 +198,18 @@ public Builder writeParallelism(int newWriteParallelism) {
         }
       }
 
-      IcebergStreamWriter<RowData> streamWriter = createStreamWriter(table, tableSchema);
+      // Find out the equality field id list based on the user-provided equality field column names.
+      List<Integer> equalityFieldIds = Lists.newArrayList();
+      if (equalityFieldColumns != null && equalityFieldColumns.size() > 0) {
+        for (String column : equalityFieldColumns) {
+          org.apache.iceberg.types.Types.NestedField field = table.schema().findField(column);
+          Preconditions.checkNotNull(field, "Missing required equality field column '%s' in table schema %s",
+              column, table.schema());
+          equalityFieldIds.add(field.fieldId());
+        }
+      }
```

Comment:
Why not do this conversion in `equalityFieldColumns` and keep the column ids in the builder instead of the source column names?

Reply:
Because the `FlinkSink` is an API which will be exposed to flink's DataStream users,   the concept of equality field id is harder to understand for those flink users.  Equality field column names will be more friendly.